# Does captivity change the microbial community?
```{r data_acclim, comment="", message=FALSE, warning=FALSE, echo=FALSE}
load("data/objects_2803.Rdata")
load("data/beta_08032025.Rdata")
load("data/alpha_08032025.Rdata")
```

## Alpha diversity

```{r filter_data_accli, comment="", message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))%>%
  filter(time_point %in% c("Acclimation", "Wild") ) 
```
```{r alpha_div_plot_1_accli, comment="", message=FALSE, warning=FALSE, fig.height=12, fig.width=12, fig.fullwidth=TRUE}
label_map <- c(
  "richness" = "Species Richness",
  "neutral" = "Neutral Diversity",
  "phylogenetic" = "Phylogenetic Diversity")

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = "sample") %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic")),
  time_point = factor(time_point, levels = c("Wild", "Acclimation"))) %>%
  filter(time_point %in% c("Wild", "Acclimation")) %>% 
  ggplot(aes(y = value, x = time_point, color=Population, fill=Population)) +
  geom_boxplot(width = 0.5,alpha=0.5, outlier.shape = NA, show.legend = FALSE) +
  geom_jitter(width = 0.2,show.legend = FALSE) +
  scale_color_manual(values=c('#008080', "#d57d2c")) +
  scale_fill_manual(values=c('#008080', "#d57d2c")) +
  facet_grid(metric ~ Population, scales = "free_y", labeller = labeller(metric = label_map, type = label_map))+
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 14),
    strip.text = element_text(size = 18, lineheight = 0.6),
      ) +
  ylab("Alpha diversity")
```

***Richness***
```{r LMM_accli, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glmer.nb(richness ~ Population*time_point+(1|individual), data = alpha_div_meta)
summary(Modelq0GLMMNB)
```

***Neutral***
```{r LMM_neutral_accli, comment="", message=FALSE, warning=FALSE}
Model_neutral <- nlme::lme(fixed = neutral ~ time_point*Population, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_neutral)
```

***Phylogenetic***
```{r LMM_phylo_accli, comment="", message=FALSE, warning=FALSE}
Model_phylo <- nlme::lme(fixed = phylogenetic ~ time_point*Population, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_phylo)
```

## Beta diversity
```{r filter_data1_accli, comment="", message=FALSE, warning=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(time_point %in% c("Acclimation", "Wild")) %>% 
  select(Tube_code) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(time_point %in% c("Acclimation", "Wild"))
```

***Richness***
```{r permanova_richness_accli, comment="", message=FALSE, warning=FALSE}
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
betadisper(richness, subset_meta$time_point) %>% permutest(.) 
```

```{r permanova_richness_accli_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ time_point*Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(richness))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

***Neutral***
```{r permanova_neutral_accli, comment="", message=FALSE, warning=FALSE}
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])

betadisper(neutral, subset_meta$time_point) %>% permutest(.)
```

```{r permanova_neutral_accli_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ time_point*Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(neutral))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(neutral))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

***Phylogenetic***
```{r permanova_phylo_accli, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$time_point) %>% permutest(.) 
```

```{r permanova_phylo_accli_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(phylo ~ time_point*Population,
        data = subset_meta %>% arrange(match(Tube_code,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(phylo))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

***dbRDA***
```{r rda_accli_wdil, comment="", message=FALSE, warning=FALSE}

#Richness
cca_ord <- capscale(formula = richness ~ subset_meta$time_point* subset_meta$Population)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta$time_pointWild" = "Wild",
                                 "subset_meta$PopulationWarm"="Warm population",
                                 "subset_meta$time_pointWild:subset_meta$PopulationWarm"="Interaction Wam population")

CAP_df %>%
  group_by(Population, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=Population,shape = time_point)) +
  scale_color_manual(values=c('#008080', "#d57d2c")) +
  scale_fill_manual(values=c('#00808050', "#d57d2c50")) +
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()

#Neutral
cca_ord <- capscale(formula = neutral ~ subset_meta$time_point* subset_meta$Population)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta$time_pointWild" = "Wild",
                                 "subset_meta$PopulationWarm"="Warm population",
                                 "subset_meta$time_pointWild:subset_meta$PopulationWarm"="Interaction Wam population")

CAP_df %>%
  group_by(Population, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=Population,shape = time_point)) +
  scale_color_manual(values=c('#008080', "#d57d2c")) +
      scale_fill_manual(values=c('#00808050', "#d57d2c50"))+
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()


#Phylogenetic
cca_ord <- capscale(formula = phylo ~ subset_meta$time_point* subset_meta$Population)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta$time_pointWild" = "Wild",
                                 "subset_meta$PopulationWarm"="Warm population",
                                 "subset_meta$time_pointWild:subset_meta$PopulationWarm"="Interaction Wam population")

CAP_df %>%
  group_by(type, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=Population,shape = time_point)) +
  scale_color_manual(values=c('#008080', "#d57d2c")) +
  scale_fill_manual(values=c('#00808050', "#d57d2c50")) +
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()
```

## Differences in MAGs

### Cold population

#### Structural zeros

```{r struct_zero_accli, comment="", message=FALSE, warning=FALSE}
wild_samples <- sample_metadata %>%
  filter(Population == "Cold" & time_point == "Wild") %>% 
  dplyr::select(sample) %>%
  pull()

acclimation_samples <- sample_metadata %>%
  filter(Population == "Cold" & time_point == "Acclimation") %>% 
  dplyr::select(sample) %>% pull()

subset_meta <- sample_metadata %>%
  filter(Population == "Cold" & time_point %in% c("Acclimation","Wild"))

structural_zeros <- genome_counts_filt %>% 
  select(one_of(c("genome",subset_meta$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_wild = all(c_across(all_of(wild_samples)) == 0)) %>% 
   mutate(all_zeros_acclimation= all(c_across(all_of(acclimation_samples)) == 0)) %>% 
   mutate(average_wild = mean(c_across(all_of(wild_samples)), na.rm = TRUE)) %>% 
   mutate(average_acclimation = mean(c_across(all_of(acclimation_samples)), na.rm = TRUE)) %>% 
   filter(all_zeros_wild == TRUE || all_zeros_acclimation==TRUE)  %>% 
   mutate(present = case_when(
      all_zeros_wild & !all_zeros_acclimation ~ "acclimation",
      !all_zeros_wild & all_zeros_acclimation ~ "wild",
      !all_zeros_wild & !all_zeros_acclimation ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "acclimation", average_wild, average_acclimation)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
```

***Wild***
```{r structural_zeros_accli_1, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="wild") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 

structural_zeros %>% 
  filter(present=="wild")%>% 
  select(1,5:10)
```

***Acclimation***
```{r structural_zeros_wild, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="acclimation") %>% 
  select(1,5:10)
```

***Community elements differences in structural zeros***
```{r gift_struc_acclim_wild, comment="", message=FALSE, warning=FALSE}
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% structural_zeros$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

sample_sub <- sample_metadata %>%
  filter(Population == "Cold" & time_point %in% c("Acclimation", "Wild"))

genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",sample_sub$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
  column_to_rownames(., "genome") 
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
```

```{r comunity_elem_acclim_wild_struc, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(8,9)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_struc_acclim_wild, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
  pivot_longer(-c(sample, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  left_join(., uniqueGIFT_db,by=join_by("trait"=="Code_element") )

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(group_color = ifelse(Difference <0, "Wild","Acclimation")) 
```

```{r commun_wilcox_elem_plot_acclim_wild, comment="", message=FALSE, warning=FALSE, fig.height=12, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#00808050", '#008080')) +
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  labs(fill="time_point")
```


#### Differential abundance analysis: composition
```{r phylo_acclim_wild, comment="", message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  filter(Population == "Cold" & time_point %in% c("Wild", "Acclimation") )%>% 
  column_to_rownames("sample") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```
```{r ancom_rand_accl_wild, comment="", message=FALSE, warning=FALSE}
ancombc_wild_accl_2803 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point",
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r save_ancombc1_wild_accli, warning=FALSE, comments="", message=FALSE}
save(ancombc_wild_accl_2803, 
     file = "data/ancombc_wild_accl_2803.Rdata")
```

```{r load_ancombc_wild_accli, warning=FALSE, comments="", message=FALSE, eval=FALSE}
load("data/ancombc_wild_accl_2803.Rdata")
```

```{r ancom_rand1_wild_accli, comment="", message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancombc_wild_accl_2803$res %>%
  dplyr::select(taxon, lfc_time_pointWild, p_time_pointWild) %>%
  filter(p_time_pointWild < 0.05) %>%
  dplyr::arrange(p_time_pointWild) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointWild)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy1_wild_accli, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_time_pointWild, y=forcats::fct_reorder(genome,lfc_time_pointWild), fill=phylum)) + 
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```
MAGs enriched in acclimation on the left side and in wild on the right side



***Phyla of the significant MAGs***

Wild
```{r acclim_sign_features_wild_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild>0)  %>% 
  select(phylum, genus, species)
```

Acclimation
```{r acclim_sign_features_accli_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild<0)  %>% 
  select(phylum, genus, species)
```

#### Differences in the functional capacity
```{r gift_acclim_antib_wild_accli, comment="", message=FALSE, warning=FALSE}
sample_sub <- sample_metadata %>%
  filter(Population == "Cold" & time_point %in% c("Acclimation", "Wild"))
genome_counts_filt_sub <- genome_counts_filt %>%
  select(one_of(c("genome",sample_sub$sample)))

GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filt_sub$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_filt_sub_row <- genome_counts_filt_sub %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_filt_sub_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_filt_sub_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_filt_sub_row,GIFT_db)
```

##### Community elements differences:

***MCI***
```{r gitfs_elem_accli_wild, comment="", message=FALSE, warning=FALSE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value))%>% 
  mutate(MCI = str_c(round(MCI, 3), "±", round(sd, 3))) %>% 
  column_to_rownames(., "time_point") %>% 
  select(MCI)

MCI <- GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ time_point, data=MCI)
```

```{r comunity_elem_acclim_wild, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(8,9)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_acclim_wild, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,time_point), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(group_color = ifelse(Difference <0, "Wild","Acclimation")) 
difference_table
```

##### Community functions differences

***MCI***
```{r gitfs_accli_wild, message=FALSE, warning=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value)) %>% 
  mutate(MCI = str_c(round(MCI, 3), "±", round(sd, 3))) %>% 
  column_to_rownames(., "time_point") %>% 
  select(MCI)

MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ time_point, data=MCI)
```

```{r comunity_func_Acci_wild, comment="", message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(8,9)], by="sample")
```

```{r commun_wilcox_func_wild_accli, comment="", message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

significant_functional <- function_gift %>%
  pivot_longer(-c(sample,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))
```
```{r  function_sig_wild_accli, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_t <- function_gift  %>% 
  select(-time_point)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt <- subset(function_gift_t, trait %in% significant_functional$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

uniqueGIFT_func<- unique(GIFT_db[c(3,5)])

difference_table <- function_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Code_function") %>%
  left_join(.,uniqueGIFT_func,by = join_by(Code_function == Code_function)) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(time_point = ifelse(Difference <0, "Wild","Acclimation")) 

difference_table%>% 
  arrange(-Difference) %>% 
  tt() 
```

### Warm population
#### Structural zeros

```{r struct_zero_accli_warm, comment="", message=FALSE, warning=FALSE}
wild_samples_warm <- sample_metadata %>%
  filter(Population == "Warm" & time_point == "Wild") %>% 
  dplyr::select(sample) %>%
  pull()

acclimation_samples_warm <- sample_metadata %>%
  filter(Population == "Warm" & time_point == "Acclimation") %>% 
  dplyr::select(sample) %>% pull()

sample_sub <- sample_metadata %>%
  filter(Population == "Warm" & time_point %in% c("Acclimation", "Wild"))
 
structural_zeros <- genome_counts_filt %>% 
  select(one_of(c("genome",sample_sub$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_wild_warm = all(c_across(all_of(wild_samples_warm)) == 0)) %>% 
   mutate(all_zeros_acclimation_warm= all(c_across(all_of(acclimation_samples_warm)) == 0)) %>% 
   mutate(average_wild_warm = mean(c_across(all_of(wild_samples_warm)), na.rm = TRUE)) %>% 
   mutate(average_acclimation_warm = mean(c_across(all_of(acclimation_samples_warm)), na.rm = TRUE)) %>% 
   filter(all_zeros_wild_warm == TRUE || all_zeros_acclimation_warm==TRUE)  %>% 
   mutate(present = case_when(
      all_zeros_wild_warm & !all_zeros_acclimation_warm ~ "acclimation",
      !all_zeros_wild_warm & all_zeros_acclimation_warm ~ "wild",
      !all_zeros_wild_warm & !all_zeros_acclimation_warm ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "acclimation", average_wild_warm, average_acclimation_warm)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
```

***Wild***
```{r structural_zeros_accli_1_warm, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="wild") %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count)) 
structural_zeros %>% 
  filter(present=="wild") %>% 
  select(1,5:10)
```

***Acclimation***
```{r structural_zeros_wild_warm, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="acclimation") %>% 
  select(1,5:10)
```

***Community elements differences in structural zeros***
```{r gift_struc_acclim_wild_warm, comment="", echo=FALSE, message=FALSE, warning=FALSE}
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% structural_zeros$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",sample_sub$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
  column_to_rownames(., "genome") 
GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
```

```{r comunity_elem_acclim_wild_struc_warm, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(8,9)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_struc_acclim_wild_warm, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,time_point), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(group_color = ifelse(Difference <0, "Wild","Acclimation")) 
difference_table
```

***Differential MAG abundance***
```{r phylo_acclim_wild_warm, comment="", message=FALSE, warning=FALSE}
phylo_samples_warm <- sample_metadata %>%
  filter(Population == "Warm" & time_point %in% c("Wild", "Acclimation") )%>% 
  column_to_rownames("sample") %>% 
  sample_data()
phylo_genome_warm <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples_warm)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy_warm <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome_warm)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered_warm <- phyloseq(phylo_genome_warm, phylo_taxonomy_warm, phylo_samples_warm)
```
```{r ancom_rand_accl_wild_warm, comment="", message=FALSE, warning=FALSE}
ancombc_wild_accl_warm_2803 = ancombc2(data = physeq_genome_filtered_warm, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point",
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r save_ancombc1_wild_accli_warm, warning=FALSE, comments="", message=FALSE}
save(ancombc_wild_accl_warm_2803, 
     file = "data/ancombc_wild_accl_warm_2803.Rdata")
```

```{r load_ancombc_wild_accli_warm, warning=FALSE, comments="", message=FALSE, eval=FALSE}
load("data/ancombc_wild_accl_warm_2803.Rdata")
```

```{r ancom_rand1_wild_accli_warm, comment="", message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered_warm@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancombc_wild_accl_warm_2803$res %>%
  dplyr::select(taxon, lfc_time_pointWild, p_time_pointWild) %>%
  filter(p_time_pointWild < 0.05) %>%
  dplyr::arrange(p_time_pointWild) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointWild)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy1_wild_accli_warm, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_time_pointWild, y=forcats::fct_reorder(genome,lfc_time_pointWild), fill=phylum)) + 
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```
MAGs enriched in acclimation on the left side and in wild on the right side



***Phyla of the significant MAGs***

Acclimation
```{r acclim_sign_features_accli_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild<0)  %>% 
  select(phylum, genus, species)
```

Wild
```{r acclim_sign_features_wild_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_time_pointWild>0)  %>% 
  select(phylum, genus, species)
```

#### Differences in the functional capacity
```{r gift_acclim_antib_wild_accli_warm, comment="", message=FALSE, warning=FALSE}
sample_sub <- sample_metadata %>%
  filter(Population == "Warm" & time_point %in% c("Acclimation", "Wild"))
genome_counts_sub <- genome_counts_filt %>%
    select(one_of(c("genome",sample_sub$sample)))

GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_sub$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_sub_row <- genome_counts_sub %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_sub_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_sub_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_sub_row,GIFT_db)
```

##### Community elements differences:

***MCI***
```{r gitfs_elem_accli_wild_warm, comment="", message=FALSE, warning=FALSE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value))%>% 
  mutate(MCI = str_c(round(MCI, 3), "±", round(sd, 3))) %>% 
  column_to_rownames(., "time_point") %>% 
  select(MCI)

MCI <- GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ time_point, data=MCI)
```

```{r comunity_elem_acclim_wild_warm, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(8,9)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_acclim_wild_warm, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,time_point), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(group_color = ifelse(Difference <0, "Wild","Acclimation")) 
```

```{r commun_wilcox_elem_plot_acclim_wild_warm, comment="", message=FALSE, warning=FALSE, fig.height=12, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
#  geom_point(size=4) + 
  scale_fill_manual(values=c("#d57d2c50","#d57d2c")) +
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  labs(fill="time_point")
```

##### Community functions differences

***MCI***
```{r gitfs_accli_wild_warm, message=FALSE, warning=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(time_point) %>%
  summarise(MCI = mean(value), sd = sd(value))%>% 
  mutate(MCI = str_c(round(MCI, 3), "±", round(sd, 3))) %>% 
  column_to_rownames(., "time_point") %>% 
  select(MCI)

MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ time_point, data=MCI)
```

```{r comunity_func_Acci_wild_warm, comment="", message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(8,9)], by="sample")
```

```{r commun_wilcox_func_wild_accli_warm, comment="", message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

significant_functional <- function_gift %>%
  pivot_longer(-c(sample,time_point), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))
```
```{r  function_sig_wild_accli_warm, comment="", echo=FALSE, message=FALSE, warning=FALSE}
function_gift_t <- function_gift  %>% 
  select(-time_point)  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

function_gift_filt <- subset(function_gift_t, trait %in% significant_functional$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(8,9)], by = join_by(sample == sample))

uniqueGIFT_func<- unique(GIFT_db[c(3,5)])

difference_table <- function_gift_filt %>%
  select(-sample) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Code_function") %>%
  left_join(.,uniqueGIFT_func,by = join_by(Code_function == Code_function)) %>% 
  mutate(Difference=Acclimation-Wild)%>% 
  mutate(time_point = ifelse(Difference <0, "Wild","Acclimation")) 

difference_table%>% 
  arrange(-Difference) %>% 
  tt() 
```